test/lit/exec/strings.wast

```diff
@@ -476,24 +476,6 @@
       (string.const "five!")
     )
   )
-
-  ;; CHECK:      [fuzz-exec] calling extern
-  ;; CHECK-NEXT: [fuzz-exec] note result: extern => string("string")
-  (func $extern (export "extern") (result externref)
-    (extern.convert_any
-      (string.const "string")
-    )
-  )
-
-  ;; CHECK:      [fuzz-exec] calling extern-intern
-  ;; CHECK-NEXT: [fuzz-exec] note result: extern-intern => string("string")
-  (func $extern-intern (export "extern-intern") (result anyref)
-    (any.convert_extern
-      (extern.convert_any
-        (string.const "string")
```

Comment:
It might still be useful to test `any.convert_extern` of a string.

Reply:
I don't follow - convert it from what to what?

In the new model a string is an externref like a JS object is an externref: it doesn't have a wasm representation under `any`.

Reply:
`any.convert_extern` and `extern.convert_any` are infallible; all `any` and `extern` values, including strings, can be converted to the other representation.

Reply:
But what is the "other representation" of string? I.e. what are you saying that `any.convert_extern` of a string should return - what actual type?

Reply:
...so it *is* possible to make a value that has (exactly) type anyref? Hence `(ref exact any)` is *not* uninhabitable?

Reply:
I think the easiest fix in the interpreter will be to update the interpretation of casts to have them always fail if the cast target type is an exact reference to an abstract heap type. That seems simpler than actually realizing a hidden subtype of `any` in our type system.

Reply:
That might work in the interpreter but I think it is dangerous in the optimizer as well. Seems safer to just add an `anystring` type (or keep the existing one as that).

How about leaving this for later? We don't urgently need this functionality (no known codebase converts string to any, in practice).

Reply:
Sure, I guess we can leave it for later. I need to update the interpretation of casts to handle exact types at some point anyway.

Reply:
The fuzzer hits this, so we do need to resolve the matter in this PR somehow. I'm not sure it is easy to work around it in the fuzzer.

Reply:
We already support `Literal`s with type externref that can be unwrapped to access the underlying `any` value. This is the same thing in reverse.

Reply:
Do/should exact casts to externref work on such values?